modules/dmrpp_module/Chunk.cc

```diff
@@ -1228,6 +1229,10 @@ void Chunk::read_chunk() {
     if (d_is_read)
         return;
 
+    if (get_data_url() != nullptr) {
+        BES_PROFILE_TIMING(string("Read chunk - ") + get_data_url()->str() + string(" - size: ") + ::to_string(get_size()) + string(" - offset: ") + ::to_string(get_offset()));
```

Comment:
Supports item 6.1 (see description)

Reply:
My understanding was that the mission was to profile SuperChunk performance.  Placing a timer here will mean that it gets triggered thousands of times, even tens of thousands of time per request. I believe that this  will flood the logs and have noticeable performance consequences.

Reply:
Also - I am pretty sure that the way the code is written is to put all the chunks into a SuperChunk. (Some SuperChunk represent only a single Chunk) And then we read the SuperChunk(s). Once the SuperChunk has read the constituent Chunk data the constituent Chunks are all marked as read, and the individual Chunk::read() methods are not actually called.  I think...

Reply:
> Placing a timer here will mean that it gets triggered thousands of times, even tens of thousands of times per request depending on the granule. I believe that this will flood the logs and have noticeable performance consequences.

Yikes! Great point, removing. 

Why did I add this in the first place? In one of my test cases, where I have a single constraint expression, I never hit the SuperChunk data retrieval calls (the checkpoints added to SuperChunk.cc). I'll try to hunt down a less-noisy checkpoint site for that case.

Reply:
(Also, fwiw, you are correct about 

> And it seems like I have contradicted myself: If the Chunk::read() methods are never called then why not instrument them? Indeed. Maybe you should leave and see what shakes out in testing.

They are called, and called a lot! It was one of the things I'd flagged to ask you about in person, even before you'd made this comment on the PR. :) )

Reply:
Okay, actually, not adding any non-superchunk checkpoints unless we intentionally decide otherwise.